### PR TITLE
Add go.mod and pin dependencies

### PR DESCRIPTION
**Pull request by ashokponkumar**
Signed-off-by: Ashok Pon Kumar <ashokponkumar@gmail.com>

### CODE REVIEW

**Comment by vbatts** [comment]
not a bad idea, but it fails travis CI

**Comment by vbatts** [comment]
@ashokponkumar could you rebase?

File: go.sum [hunk 1]
```diff
@@ -0,0 +1,241 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (vbatts) [inline go.sum:1]: this go.sum file does not even need to be checked in

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module github.com/opencontainers/image-spec
```
Comment (vbatts) [inline go.mod:1]: after some discussion on the call, I'm not sure that we want to have a single `go.mod` at the top level of the project, but instead to have a `go.mod` in any immediate subdirectory that contains golang source.

The reasoning behind this is: if say `./schema/` brings in a dependency that is wholly not needed if someone only needs to import `./specs-go/`, importers need only fetch `specs-go/go.mod` related deps (which is likely always none).

We just did similar on the distribution-spec

**Review by vbatts — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (ashokponkumar) [reply go.sum]: Its generally the practise to checkin go.sum too. But if we don't want for this repo, we can remove it. (Ex: https://github.com/kubernetes/kubernetes/blob/master/go.sum)

File: go.mod (hunk 2 above)
Comment (ashokponkumar) [reply go.mod]: Sure. I have added for specs-go. Adding to schema to creates some ambiguous import issue.

```ashoks-mbp:schema ashokponkumar$ go mod tidy
go: finding module for package github.com/opencontainers/image-spec/specs-go/v1
go: found github.com/opencontainers/image-spec/specs-go/v1 in github.com/opencontainers/image-spec v1.0.1
github.com/opencontainers/image-spec/schema: ambiguous import: found package github.com/opencontainers/image-spec/schema in multiple modules:
	github.com/opencontainers/image-spec/schema (/Users/ashokponkumar/go/src/github.com/ashokponkumar/image-spec/schema)
	github.com/opencontainers/image-spec v1.0.1 (/Users/ashokponkumar/go/pkg/mod/github.com/opencontainers/image-spec@v1.0.1/schema)```

Comment (vbatts) [reply go.mod]: oh dang. Just did some research on this, and it's a go modules mess ... and sounds like what @tianon mentioned during the image-spec grooming call with having to do special git tags also.

I found that the popular import `github.com/ugorji/go/codec` went through this when they brought in go modules.
https://github.com/ugorji/go/issues/299
They ended up having to add circular referencing go.mod files, based on tag versions, and also to make multiple tags for each of the sub-directories as their own modules https://github.com/ugorji/go/tags

I'm now wondering if this is all worth it, or is it just simpler to keep the go.mod file at the topic level directory like you began with :-\

@opencontainers/image-spec-maintainers thoughts?

(p.s. i'm curious why the we didn't hit this in the distribution-spec?)

Comment (ashokponkumar) [reply go.mod]: The reason the issue is not there in distribution-spec is that, it has only one go module in the repo.

**Comment by vbatts** [comment]
thank you @ashokponkumar for your PR! and congrats on your first commit!

**Comment by ashokponkumar** [comment]
Thanks @vbatts for taking it through the process and getting it merged.

**Pull request closed without merging**